image_resources.cpp

```diff
@@ -130,5 +132,225 @@ bool ImageResource::resIDHasDescriptor(uint16_t resID)
+  klass->className = getUnicodeString();
+  klass->meta = parseDescrVariable();
+  return klass;
+}
+
+std::unique_ptr<OSType> Decoder::parseEnumeratedType() {
+  auto enum_ = std::make_unique<OSTypeEnum>();
+  enum_->typeID = parseDescrVariable();
+  enum_->enumValue = parseDescrVariable();
+  return enum_;
+}
+
+std::unique_ptr<OSType> Decoder::parseAliasType() {
+  const uint32_t length = read32();
+  m_file->seek(m_file->tell() + length);
+  return std::make_unique<OSTypeAlias>();
+}
+
+std::unique_ptr<OSType> Decoder::parseOsTypeVariable() {
+    const OSTypeClassMetaType descMeta = parseDescrVariable();
+    if (descMeta.keyClassID == 0) {
+      //return std::make_unique<OSTypeDescriptor>(descMeta.name);
+      return std::make_unique<OSTypeDescriptor>();
+    }
+
+    const uint32_t osTypeInt = read32();
+    assert(is_valid_class_type(osTypeInt));
+    const OSTypeKey osType = static_cast<OSTypeKey>(osTypeInt);
+
+    switch (osType) {
+    case OSTypeKey::GlobalObject:
+    case OSTypeKey::Descriptor:
+      return parseDescriptor();
+    case OSTypeKey::Reference:
+      return parseReferenceType();
+    case OSTypeKey::List:
+      return parseListType();
+    case OSTypeKey::Double:
+      return std::make_unique<OSTypeDouble>((double)read64());
+    case OSTypeKey::UnitFloat: {
+      const uint32_t unit = read32();
+      const double value = read64();
+      assert(is_valid_unit_float(unit));
+      return std::make_unique<OSTypeUnitFloat>(
+          static_cast<OSTypeUnitFloat::Unit>(unit), value);
+    }
+    case OSTypeKey::String:
+      return std::make_unique<OSTypeString>(getUnicodeString());
+    case OSTypeKey::Enumerated:
+      return parseEnumeratedType();
+    case OSTypeKey::Long:
+      return std::make_unique<OSTypeInt>(read32());
+    case OSTypeKey::LargeInteger:
+      return std::make_unique<OSTypeLargeInt>(read64());
+    case OSTypeKey::Boolean:
+      return std::make_unique<OSTypeBoolean>((bool)read8());
+    case OSTypeKey::GlobalClass:
+    case OSTypeKey::ClassType:
+      return parseClassType();
+    case OSTypeKey::Alias:
+      return parseAliasType();
+    case OSTypeKey::RawData:
+      auto data = std::make_unique<OSTypeRawData>();
+      data->value.resize(descMeta.name.size());
+      std::copy(descMeta.name.begin(), descMeta.name.end(),
+                data->value.begin());
+      return data;
+    }
+    return nullptr;
+}
+
+std::unique_ptr<OSType> Decoder::parseDescriptor() {
+  auto desc = std::make_unique<OSTypeDescriptor>();
+  desc->descriptorName = getUnicodeString();
+  desc->classId = parseDescrVariable();
+
+  auto const nDescriptors = read32();
+  assert(nDescriptors > 0);
+  desc->descriptors.reserve(nDescriptors);
+
+  for (int i = 0; i < nDescriptors; ++i) {
+    desc->descriptors.emplace_back(parseOsTypeVariable());
+  }
+
+  return desc;
+}
+
+std::wstring Decoder::getUnicodeString() {
+  auto const length = read32();
+  std::wstring str{};
```

Comment:
Don't use `{}` to initialize instances of objects that are default constructed anyway:
```suggestion
  std::wstring str;
```
As a side note: We don't use `{}`. For other kind of variables we can just use the value explicitly. E.g. `int a = 0;`.